anvil/src/cmd.rs

```diff
@@ -84,8 +84,14 @@ pub struct NodeArgs {
     pub no_mining: bool,
 
     /// The host the server will listen on.
-    #[clap(long, value_name = "IP_ADDR", env = "ANVIL_IP_ADDR", help_heading = "Server options")]
-    pub host: Option<IpAddr>,
+    #[clap(
+        long,
+        value_name = "IP_ADDR",
+        env = "ANVIL_IP_ADDR",
+        default_value = "127.0.0.1",
+        help_heading = "Server options"
+    )]
+    pub host: Vec<IpAddr>,
```

Comment:
Sadly this would be a big breaking change on Anvil, and ideally we'd avoid this—we'd just enable listening on all interfaces by default (see https://users.rust-lang.org/t/how-to-listen-on-all-interfaces-with-tcplistener-solved/12269)

Reply:
Hi, @Evalir maybe I didn't explain the solution very well. I tried to not introduce any breaking changes. 
The following commands work the same way they worked before:
`anvil` - makes it listen on 127.0.0.1:8545
`env ANVIL_IP_ADDR="1.1.1.1" anvil` - makes it listen on 1.1.1.1:8545
`anvil --host 1.1.1.1` - makes it listen on 1.1.1.1:8545
etc.
but now you can also add additional hosts if needed
`anvil --host :: --host 127.0.0.1 --host 1.1.1.1` - makes it listen on [::]:8545, 127.0.0.1:8545, 1.1.1.1:8545 

The limitations in the current implementation are:
- you can't add more than one host when using the environment variable ANVIL_IP_ADDR
- all hosts will listen on the same port. This can be fixed in a clean way only by introducing a breaking change

Btw I wrote test about this, but I didn't push it, because it's testing only the _clap_'s behavior.
If you wish I can do it.

```diff
+                                                                                                                               │1803920 zerohero  20   0 1131,2g 218068 118492 S   0,0   0,3   0:07.64 brave
+    #[test]                                                                                                                    │3852120 zerohero  20   0 1131,1g 197328 121664 S   1,5   0,3   0:38.69 brave
+    fn can_parse_host() {                                                                                                      │ 130854 zerohero  20   0 1131,1g 192300 117612 S   0,0   0,3   0:04.43 brave
+        let args = NodeArgs::parse_from(["anvil"]);                                                                            │1805505 zerohero  20   0 1131,1g 189996 111580 S   0,0   0,3   0:12.79 brave
+        assert_eq!(args.host, vec![IpAddr::V4(Ipv4Addr::LOCALHOST)]);                                                          │3066158 zerohero  20   0 1131,1g 189056  96100 S   0,0   0,3   0:29.47 brave
+                                                                                                                               │1804908 zerohero  20   0 1131,1g 188652 111264 S   0,0   0,3   0:12.34 brave
+        let args = NodeArgs::parse_from([                                                                                      │4031496 zerohero  20   0 1133,1g 187688 113996 S   0,0   0,3   0:02.55 brave
+            "anvil", "--host", "::1", "--host", "1.1.1.1", "--host", "2.2.2.2",                                                │3297574 zerohero  20   0 1131,2g 186376  96336 S   0,0   0,3   0:29.72 brave
+        ]);                                                                                                                    │3926891 zerohero  20   0 1131,1g 182908 114688 S   0,0   0,3   0:02.89 brave
+        assert_eq!(                                                                                                            │1804909 zerohero  20   0 1131,1g 179244 113544 S   0,0   0,3   0:06.86 brave
+            args.host,                                                                                                         │3822023 zerohero  20   0 1131,0g 179188 115236 S   0,0   0,3   0:01.23 brave
+            ["::1", "1.1.1.1", "2.2.2.2"].map(|ip| ip.parse::<IpAddr>().unwrap()).to_vec()                                     │3641763 zerohero  20   0 1131,1g 179056 115696 S   0,0   0,3   0:01.12 brave
+        );                                                                                                                     │1996379 zerohero  20   0 1133,2g 178328  84036 S   0,0   0,3   0:36.43 brave
+                                                                                                                               │2289036 zerohero  20   0 1131,1g 176592 117696 S   0,0   0,3   0:07.60 brave
+        env::set_var("ANVIL_IP_ADDR", "1.1.1.1");                                                                              │2125889 zerohero  20   0 1133,2g 173768 101704 S   0,0   0,3   3:22.59 brave
+        let args = NodeArgs::parse_from(["anvil"]);                                                                            │3439514 zerohero  20   0 1131,1g 171528 112280 S   0,0   0,3   0:02.51 brave
+        assert_eq!(args.host, vec!["1.1.1.1".parse::<IpAddr>().unwrap()]);                                                     │ 157707 zerohero  20   0 1122,0g 169912  72620 S   0,0   0,3  18:45.37 code
+    }   
```

Reply:
Could you push the tests please? That'd be helpful!

Reply:
it's done

Reply:
Just updated the description to include the content from my first comment